### ISSUE TITLE
Validation on Lazy foreign key references for a subclassed table appears to validate against the base clase table rather than the subclass

### ISSUE THREAD

**Issue by scott2b** [piccolo-orm/piccolo#521]
I have a class with an "owner" property that is a foreign key with a LazyTableReference to a User. The User class is a subclass of piccolo's BaseUser, for which I have defined the tablename to be "users":

```
class User(BaseUser, tablename="users"):
     ...
```

```
class Token(Table, tablename="tokens"):
    owner = ForeignKey(references=LazyTableReference(
                table_class_name="User", module_path="mymodule.path"
            ))
    ...
```

When I try to create a new Token instance with a User reference as the owner, I get the following error referencing the wrong table, being that of the base class rather than the instance provided:

```
DETAIL:  Key (owner)=(2) is not present in table "piccolo_user".
```

**Comment by scott2b** [comment]
For what it is worth: I tried a workaround by creating a new instance without the owner reference, then setting the owner and re-saving the instance, however I get the same foreign key violation with an apparent attempt to reference the table associated with the base user class.

**Comment by scott2b** [comment]
Ok. Resolved this. This was utterly and completely my own doing. Through somehow messing variously with the piccolo configs and/or migrations, I got myself into a place where the actual foreign key constrain in the database was pointing to the piccolo base user instead of my subclass user table. The piccolo_user table really should not even be there, let alone the constraint .. it's just a historical mess from my various changes and experiments. There is not a real issue here. Once I fixed the references in the database this all appears to be working now. Sorry for the false alarm.
